### ISSUE TITLE
Run resnet50_trainer.py error between 2 machines using mpirun

### ISSUE THREAD

**Issue by bukiya** [facebookarchive/caffe2#2459]
I try to run resnet50_trainer.py between 2 machines using mpirun, but I get the following error messages. How to fix the error and run correctly using mpirun?

My mpirun command:
mpirun --host 192.168.2.50 -np 1 python resnet50_trainer.py --train_data /dataset/ilsvrc12_train_lmdb/ --num_epochs 10 --batch_size 16 : --host 192.168.2.51 -np 1 python resnet50_trainer.py --train_data /dataset/ilsvrc12_train_lmdb/ --num_epochs 10 --batch_size 16

Error message:
Ignoring @/caffe2/caffe2/contrib/nccl:nccl_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/contrib/gloo:gloo_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/contrib/gloo:gloo_ops_gpu as it is not a valid file.
Ignoring @/caffe2/caffe2/distributed:file_store_handler_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/distributed:redis_store_handler_ops as it is not a valid file.
E0328 10:00:55.757899  3741 init_intrinsics_check.cc:59] CPU feature avx is present on your machine, but the Caffe2 binary is not compiled with it. It means you may not get the full speed of your CPU.
E0328 10:00:55.758118  3741 init_intrinsics_check.cc:59] CPU feature avx2 is present on your machine, but the Caffe2 binary is not compiled with it. It means you may not get the full speed of your CPU.
E0328 10:00:55.758144  3741 init_intrinsics_check.cc:59] CPU feature fma is present on your machine, but the Caffe2 binary is not compiled with it. It means you may not get the full speed of your CPU.
INFO:resnet50_trainer:Running on GPUs: [0]
INFO:resnet50_trainer:Using epoch size: 1500000
INFO:data_parallel_model:Parallelizing model for devices: [0]
INFO:data_parallel_model:Create input and model training operators
INFO:data_parallel_model:Model for GPU : 0
INFO:data_parallel_model:Adding gradient operators
INFO:data_parallel_model:Add gradient all-reduces for SyncSGD
WARNING:data_parallel_model:Distributed broadcast of computed params is not implemented yet
INFO:data_parallel_model:Post-iteration operators for updating params
INFO:data_parallel_model:Calling optimizer builder function
Ignoring @/caffe2/caffe2/contrib/nccl:nccl_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/contrib/gloo:gloo_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/contrib/gloo:gloo_ops_gpu as it is not a valid file.
INFO:data_parallel_model:Add initial parameter sync
Ignoring @/caffe2/caffe2/distributed:file_store_handler_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/distributed:redis_store_handler_ops as it is not a valid file.
E0328 10:00:55.974058  1246 init_intrinsics_check.cc:59] CPU feature avx is present on your machine, but the Caffe2 binary is not compiled with it. It means you may not get the full speed of your CPU.
INFO:resnet50_trainer:Running on GPUs: [0]
INFO:resnet50_trainer:Using epoch size: 1500000
INFO:data_parallel_model:Parallelizing model for devices: [0]
INFO:data_parallel_model:Create input and model training operators
INFO:data_parallel_model:Model for GPU : 0
*** Aborted at 1522202456 (unix time) try "date -d @1522202456" if you are using GNU date ***
PC: @     0x7f33718a7a8f caffe2::ConvPoolOpBase<>::TensorInferenceForConv()
*** SIGSEGV (@0x0) received by PID 3741 (TID 0x7f339111e700) from PID 0; stack trace: ***
    @     0x7f338fe654b0 (unknown)
    @     0x7f33718a7a8f caffe2::ConvPoolOpBase<>::TensorInferenceForConv()
    @     0x7f337281240e std::_Function_handler<>::_M_invoke()
    @     0x7f33717e1224 caffe2::InferBlobShapesAndTypes()
    @     0x7f33717e2b83 caffe2::InferBlobShapesAndTypesFromMap()
    @     0x7f338aedd1be \_ZZN8pybind1112cpp_function10initializeIZN6caffe26python16addGlobalMethodsERNS_6moduleEEUlRKSt6vectorINS_5bytesESaIS7_EESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES6_IlSaIlEESt4lessISI_ESaISt4pairIKSI_SK_EEEE35_S7_JSB_SR_EJNS_4nameENS_5scopeENS_7siblingEEEEvOT_PFT0_DpT1_EDpRKT2_ENUlRNS_6detail13function_callEE1_4_FUNES19\_
    @     0x7f338aefcbf3 pybind11::cpp_function::dispatcher()
    @     0x7f3390c241e5 PyEval_EvalFrameEx
    @     0x7f3390c25c3e PyEval_EvalCodeEx
    @     0x7f3390c251f7 PyEval_EvalFrameEx
    @     0x7f3390c25c3e PyEval_EvalCodeEx
    @     0x7f3390c251f7 PyEval_EvalFrameEx
    @     0x7f3390c25c3e PyEval_EvalCodeEx
    @     0x7f3390c251f7 PyEval_EvalFrameEx
    @     0x7f3390c25c3e PyEval_EvalCodeEx
    @     0x7f3390c251f7 PyEval_EvalFrameEx
    @     0x7f3390c25c3e PyEval_EvalCodeEx
    @     0x7f3390c25d52 PyEval_EvalCode
    @     0x7f3390c46450 PyRun_FileExFlags
    @     0x7f3390c4662f PyRun_SimpleFileExFlags
    @     0x7f3390c5bfd4 Py_Main
    @     0x7f338fe50830 __libc_start_main
    @           0x400729 (unknown)
INFO:data_parallel_model:Adding gradient operators
*** Aborted at 1522202456 (unix time) try "date -d @1522202456" if you are using GNU date ***
PC: @           0x49d7e4 (unknown)
*** SIGTERM (@0x3e8000004d4) received by PID 1246 (TID 0x7f7165f66700) from PID 1236; stack trace: ***
    @     0x7f7165b5f390 (unknown)
    @           0x49d7e4 (unknown)
    @           0x49c1ac (unknown)
    @           0x499457 PyType_GenericAlloc
    @           0x4aa8c5 (unknown)
    @           0x4c15bf PyEval_EvalFrameEx
    @           0x4b9ab6 PyEval_EvalCodeEx
    @           0x4c1e6f PyEval_EvalFrameEx
    @           0x4b9ab6 PyEval_EvalCodeEx
    @           0x4c16e7 PyEval_EvalFrameEx
    @           0x4b9ab6 PyEval_EvalCodeEx
    @           0x4d55f3 (unknown)
    @           0x4a577e PyObject_Call
    @           0x4bed3d PyEval_EvalFrameEx
    @           0x4b9ab6 PyEval_EvalCodeEx
    @           0x4c1e6f PyEval_EvalFrameEx
    @           0x4b9ab6 PyEval_EvalCodeEx
    @           0x4c1e6f PyEval_EvalFrameEx
    @           0x4b9ab6 PyEval_EvalCodeEx
    @           0x4c16e7 PyEval_EvalFrameEx
    @           0x4b9ab6 PyEval_EvalCodeEx
    @           0x4c1e6f PyEval_EvalFrameEx
    @           0x4b9ab6 PyEval_EvalCodeEx
    @           0x4c1e6f PyEval_EvalFrameEx
    @           0x4b9ab6 PyEval_EvalCodeEx
    @           0x4eb30f (unknown)
    @           0x4e5422 PyRun_FileExFlags
    @           0x4e3cd6 PyRun_SimpleFileExFlags
    @           0x493ae2 Py_Main
    @     0x7f71657a4830 __libc_start_main
    @           0x4933e9 _start
    @                0x0 (unknown)
\--------------------------------------------------------------------------
mpirun noticed that process rank 0 with PID 3741 on node bruce1-G752VM exited on signal 11 (Segmentation fault).

**Comment by bukiya** [comment]
Anyone could give some advice?

**Comment by gyani91** [comment]
Hey bukiya,

I have bumped into the same error. Were you able to fix it?

I would really appreciate, if you could post the solution here.

Thank you.
